### ISSUE TITLE
Open loot

### ISSUE THREAD

**Issue by Faaaaaabs** [otland/forgottenserver#1851]
Have two corpses dead.

Open loot.. and open loot first death corpse.

**Comment by wgrriffel** [comment]
incomprehensible.

**Comment by Faaaaaabs** [comment]
It's simple! 

Kill two monster the same SQM.. and open loot, after second  die monster.

he opened the first monster killed..

**Comment by wgrriffel** [comment]
reproduced, in addition it only happen while first corpse on stack doesn't decayed yet.

**Comment by marksamman** [comment]
I'm not sure I understand this issue, but my guess is that the corpses have FORCE_USE attribute before they decay into the second state, and the one on the bottom is being chosen before the item on top. If that's the case, then it's a bug.

**Comment by wgrriffel** [comment]
@marksamman yes, it's the case, the bottom item is beeing used before the top one.

**Comment by vankk** [comment]
I can confirm this issue, it happens in last commit.

**Comment by wgrriffel** [comment]
@vankk this has nothing to do with latest commits, happens to my distro builded with f057c1498840250f9c28ddce7ac59bf8714a2bf2

**Comment by andersonfaaria** [comment]
[this is a partial copy of my comment in #1968] I noticed it a long time ago but not using two corpses, instead I've tested it with a torch and a recent corpse. If a creature dies and you throw a torch over the body and clicks to light the torch the corpse will be opened. If a creature dies above a torch and you try to open the loot it will lit the torch.

the problem is in tile.cpp, exactly in this function
Item* Tile::getUseItem() const

a provisory solution would be to check in the for loop and only returning item if it has force use but isn't a container.

**Comment by wgrriffel** [comment]
@marksamman why these corpses has forceUse flag? this flag wasnt suppose to items like ladders?

**Comment by TheSumm** [comment]
I think this issue exists because of these changes: https://github.com/otland/forgottenserver/commit/fc3c8259163111dd036b977b1a656a759a2a20cf

The client already sends the client ID of the item that is supposed to be used / looked at / moved. However,  this sprite ID is not used to determine the item to be used (unless it is a hotkey action).

**Comment by thexamx** [comment]
This issue is still present in TFS 1.3, why didnt it get committed?
Kornholi fix actually works.

**Comment by wgrriffel** [comment]
@thexamx because if a corpse drops on a tile with a ladder, it will prevent players from using ladder until corpse decays, since ladder will always be on bottom of stack that patch returns always the top of stack first, best way to this issue is removing forceUse from corpses imo.

**Comment by joseluis2g** [comment]
@wgrriffel thats otb related isnt?

**Comment by wgrriffel** [comment]
@joseluis2g yes but would need to some modifications on sources, like splash overlapping corpse

**Comment by thexamx** [comment]
If ladder etc is always the top down item then why won't we check if there is something under the corpse?
Like with rope spots. If you use rope on rope spot you get roped even if there is something on it.

**Comment by wgrriffel** [comment]
you could do this way, but it's a workaround not a fix, even worse because would be in lua, forceUse flag is there for a reason, which i really tried to understand any reason to flag forUse be on first corpses state

**Comment by thexamx** [comment]
```
for (Item* item : boost::adaptors::reverse(*items)) {
	const ItemType& itemType = Item::items[item->getID()];
	if (itemType.forceUse && !itemType.isContainer()) {
		return item;
	}
}
```
This actually fixes the issue or I missed something?

**Comment by DSpeichert** [comment]
Fixed in 5ccfce5e2807cc8cc0c4d5e4bd891105fcfcffa2 / #2393
